pyirf/interpolation/visible_edges_extrapolator.py

```diff
@@ -0,0 +1,213 @@
+
+__all__ = ["ParametrizedVisibleEdgesExtrapolator"]
+
+
+def find_visible_facets(grid_points, target_point):
+    """
+    Utility function to find all facets of a convex hull that are visible from
+    a point outside. To do so, this function constructs a triangulation containing
+    the target point and returns all facets that span a triangulation simplex with
+    it.
+
+    Parameters
+    ----------
+    grid_points: np.ndarray, shape=(N, M)
+        Grid points at which templates exist. May be one ot two dimensional.
+        Have to be sorted in accending order for 1D.
+    target_point: numpy.ndarray, shape=(1, M)
+        Value for which the extrapolation is performed (target point)
+
+    Returns
+    -------
+    visible_facets: np.ndarray, shape=(L, M)
+        L visible facets, spanned by a simplex in M-1 dimensions
+        (thus a line for M=2)
+    """
+    # Build a triangulation including the target point
+    full_set = np.vstack((grid_points, target_point))
+    triag = Delaunay(full_set)
+
+    # The target point is included in a simplex with all facets
+    # visible by it
+    simplices = triag.points[triag.simplices]
+    matches_target = np.all(simplices == target_point, axis=-1)
+    target_in_simplex = np.any(matches_target, axis=-1)
+
+    # The visible facets are spanned by those points in the matched
+    # simplices that are not the target
+    facet_point_mask = ~matches_target[target_in_simplex]
+    visible_facets = np.array(
+        [
+            triag.points[simplex[mask]]
+            for simplex, mask in zip(
+                triag.simplices[target_in_simplex], facet_point_mask
+            )
+        ]
+    )
+
+    return visible_facets
+
+
+def compute_extrapolation_weights(visible_facet_points, target_point, m):
+    """
+    Utility function to compute extrapolation weight according to [1].
+
+    Parameters
+    ----------
+    visible_facet_points: np.ndarray, shape=(L, M)
+        L facets visible from target_point
+    target_point: numpy.ndarray, shape=(1, M)
+        Value for which the extrapolation is performed (target point)
+
+    Returns
+    -------
+    extrapolation_weights: np.ndarray, shape=(L)
+        Weights for each visible facet, corresponding to the extrapolation
+        weight for the respective triangulation simplex. Weigths sum to unity.
+
+    References
+    ----------
+    .. [1] P. Alfred (1984). Triangular Extrapolation. Technical summary rept.,
+    Univ. of Wisconsin-Madison. https://apps.dtic.mil/sti/pdfs/ADA144660.pdf
+    """
+
+    angles = np.array(
+        [
+            point_facet_angle(line, target_point.squeeze())
+            for line in visible_facet_points
+        ]
+    )
+    weights = np.arccos(angles) ** (m + 1)
+
+    return weights / weights.sum()
+
+
+class ParametrizedVisibleEdgesExtrapolator(ParametrizedNearestSimplexExtrapolator):
+    """
+    Extrapolator class extending the ParametrizedNearestSimplexExtrapolator.
```

Comment:
```suggestion
    Extrapolator using blending of visible edges.
```

The extends information is given automatically.

Reply:
and is an implementation detail